parser/src/parser.rs

```diff
@@ -218,6 +218,69 @@ where
     ) -> Result<PrimitiveValue> {
         // TODO add support for OB value data length resolution
         // (might need to delegate pixel data reading to a separate trait)
+
+        // if there is PixelData of undefined length
+        if header.tag == Tag(0x7FE0, 0x0010) && header.len.is_undefined() {
+            let mut buf = [0; 8];
+            from.read_exact(&mut buf)?;
+            self.bytes_read += 8;
+
+            println!("PixelData of undefined length");
+            println!("Tag({:#06X}, {:#06X})", u16::from_le_bytes([buf[0], buf[1]]), u16::from_le_bytes([buf[2], buf[3]]));
+            let len = u32::from_le_bytes([buf[4], buf[5], buf[6], buf[7]]) as usize;
+            println!("Offset table len {}", len);
+
+            let mut offset_table = vec![0; len];
+            from.read_exact(&mut offset_table)?;
+            self.bytes_read += len as u64;
+
+            let mut offsets = vec![];
+            for i in 0..len / 4 {
```

Comment:
This logic looks suspicious, considering that `i` still iterates one by one from 0 to `len/4` without being adjusted accordingly. The table offset reads are overlapping.

You could probably create the list of offsets functionally, by chunking `offset_table` into pieces of 4. Maybe use [`Itertools#tuples`](https://docs.rs/itertools/0.8.2/itertools/trait.Itertools.html#method.tuples)?